lib/takeaway.rb

```diff
@@ -0,0 +1,32 @@
+class Takeaway
+  attr_reader :menu, :selected_items, :sum
+
+  def initialize
+    @menu = { "burritos" => 7, "tacos" => 5, "chips" => 3, "guacamole" => 2 }
+    @selected_items = []
+    @sum = 0
+  end
+
+  def order(item, quantity)
+    @selected_items << { item => quantity }
+  end
+
+  def total
+    item_totals = []
+    @selected_items.map do |item|
+      price = item.keys.join()
+      quantity = item.values.join().to_i
+      item_totals << (@menu[price] * quantity)
+    end
+    "The total for the order is £#{item_totals.sum}."
+  end
+
+  def receipt
+    @selected_items.map do |item|
+      items = item.keys
+      quantity = items.values
```

Comment:
RSpec left me the error:

```
NoMethodError:
       undefined method `values' for ["burritos"]:Array
       Did you mean?  values_at
```

It's surprising that it thinks `items` is an array. That should be a clue to finding the issue with this line.